### PR TITLE
Allow for input Ids

### PR DESCRIPTION
**Pull request by GodVenn**
Allow for optional ID input to ISAR for mission, tasks and step.
This will allow the user of ISAR (e.g. Flotilla) to track tasks and step if desired.

### CODE REVIEW

File: src/isar/apis/models/start_mission_definition.py [hunk 1]
```diff
@@ -25,6 +26,7 @@ class InspectionTypes(str, Enum):
 
 
 class StartMissionInspectionDefinition(BaseModel):
+    id: Optional[str]
```
Comment (Christdej) [inline src/isar/apis/models/start_mission_definition.py:29]: I think `Optional` parameters should be after all the mandatory parameters

File: src/isar/apis/models/start_mission_definition.py [hunk 2]
```diff
@@ -33,12 +35,14 @@ class StartMissionInspectionDefinition(BaseModel):
 
 
 class StartMissionTaskDefinition(BaseModel):
+    id: Optional[str]
```
Comment (Christdej) [inline src/isar/apis/models/start_mission_definition.py:38]: I think `Optional` parameters should be after all the mandatory parameters

File: src/isar/apis/models/start_mission_definition.py [hunk 3]
```diff
@@ -33,12 +35,14 @@ class StartMissionInspectionDefinition(BaseModel):
...
     pose: InputPose
     tag: Optional[str]
     inspections: List[StartMissionInspectionDefinition]
 
 
 class StartMissionDefinition(BaseModel):
+    id: Optional[str]
```
Comment (Christdej) [inline src/isar/apis/models/start_mission_definition.py:45]: I think `Optional` parameters should be after all the mandatory parameters

File: src/isar/models/mission/mission.py [hunk 4]
```diff
@@ -22,7 +23,7 @@ class Task:
     steps: List[STEPS]
     status: TaskStatus = field(default=TaskStatus.NotStarted, init=False)
     tag_id: Optional[str] = field(default=None)
-    id: UUID = field(default_factory=uuid4, init=False)
+    id: Union[UUID, str] = field(default_factory=uuid4, init=True)
```
Comment (Christdej) [inline src/isar/models/mission/mission.py:26]:
```suggestion
    id: str = field(default_factory=uuid4, init=True)
```
Should we change to only use str? I think Union makes for confusion and I dont see the point. Imo we just change the UUID to a str as soon as it is created, then keep it str.

File: src/robot_interface/models/mission/step.py [hunk 5]
```diff
@@ -15,7 +15,7 @@ class Step:
     Base class for all steps in a mission.
     """
 
-    id: UUID = field(default_factory=uuid4, init=False)
+    id: Union[UUID, str] = field(default_factory=uuid4, init=False)
```
Comment (Christdej) [inline src/robot_interface/models/mission/step.py:18]: Should we change to only use str? I think Union makes for confusion and I dont see the point. Imo we just change the UUID to a str as soon as it is created, then keep it str

File: src/robot_interface/telemetry/payloads.py [hunk 6]
```diff
@@ -36,8 +36,8 @@ class RobotStatusPayload:
     robot_status: RobotStatus
     current_isar_state: State
     current_mission_id: Union[UUID, int, str, None]
-    current_task_id: UUID
-    current_step_id: UUID
+    current_task_id: Union[UUID, str]
+    current_step_id: Union[UUID, str]
```
Comment (Christdej) [inline src/robot_interface/telemetry/payloads.py:40]: Should we change to only use str? I think Union makes for confusion and I dont see the point. Imo we just change the UUID to a str as soon as it is created, then keep it str

**Review by Christdej — CHANGES_REQUESTED** [review]

File: src/isar/models/mission/mission.py (hunk 4 above)
Comment (GodVenn) [reply src/isar/models/mission/mission.py:26]: Agree

File: src/robot_interface/telemetry/payloads.py (hunk 6 above)
Comment (GodVenn) [reply src/robot_interface/telemetry/payloads.py:40]: Agree

File: src/isar/apis/models/start_mission_definition.py [hunk 7]
```diff
@@ -57,17 +60,31 @@ def to_isar_mission(mission_definition: StartMissionDefinition) -> Mission:
...
                 )
                 for inspection in task.inspections
             ]
         except ValueError as e:
             raise MissionPlannerError(f"Failed to create task: {str(e)}")
+
+        if contains_duplicate_ids(items=inspection_steps):
+            raise MissionPlannerError(
+                f"Failed to create task: Duplicate step IDs are not allowed"
+            )
```
Comment (aestene) [inline src/isar/apis/models/start_mission_definition.py:73]: Do we allow duplicate ideas in steps from different tasks?

File: src/isar/apis/models/start_mission_definition.py (hunk 1 above)
Comment (GodVenn) [reply src/isar/apis/models/start_mission_definition.py]: fixed

File: src/isar/apis/models/start_mission_definition.py (hunk 2 above)
Comment (GodVenn) [reply src/isar/apis/models/start_mission_definition.py]: fixed

File: src/isar/apis/models/start_mission_definition.py (hunk 3 above)
Comment (GodVenn) [reply src/isar/apis/models/start_mission_definition.py]: fixed

File: src/isar/apis/models/start_mission_definition.py (hunk 7 above)
Comment (GodVenn) [reply src/isar/apis/models/start_mission_definition.py:73]: Yes, currently, but that was not intended, good catch.
Will fix

Comment (GodVenn) [reply src/isar/apis/models/start_mission_definition.py:74]: Fixed

**Review by Christdej — APPROVED** [review]
LGTM

**Pull request merged**